Stats revampv2 empty states for some cards

ravishanker opened the pull request:
This PR updates empty states for **Most Popular Time** and **Latest Post Summary** and **Views & Visitors** cards

| Most Popular Time | Latest Post Summary | Views & Visitors
|---|---|---|
|![Screenshot_20220524_160239](https://user-images.githubusercontent.com/990349/170021697-bd830035-dfa8-4af0-9b02-be358e948483.png) | ![Screenshot_20220524_163348](https://user-images.githubusercontent.com/990349/170021753-3469a2e4-ff6c-4043-a093-8cf6a90f6052.png)| ![Screenshot_20220526_162239](https://user-images.githubusercontent.com/990349/170429703-b022bbed-79e1-4a87-91c6-28da1c5b9418.png)|



Fixes #16617 

To test:

Setup:

- Go to App Settings (Tap on Avatar at the top right hand corner on My Site -> Find App Settings)
- Select Debug Settings
- Find StatsRevampV2FeatureConfig under Features in development enable it and restart app

Test 1:
- Launch app
- Choose a site that has no data (Preferably a newly minted site)
- Go to stats either using quick links or menu 
- Make sure Stats is on **Insights** tab if not switch to it.
- Make sure **Most Popular Time** card is displaying, if not use Stats Management to add it (scroll down if necessary)
- Ensure that it display empty data state as shown above in the first column

Test 2:
- Make sure **Latest Post Summary** card is displaying, if not use Stats Management to add it (scroll down if necessary)
- Ensure that it display empty data state as shown above in the second column

Test 3:
- Make sure **Views and Visitors** card is displaying, if not use Stats Management to add it (scroll down if necessary)
- Ensure that it display empty data state as shown above in the third column
- Click on the icon, check that it takes to top tips page

NOTE:
- Create post is an existing action on **Latest Post Summary** card and is kept as it is along with other actions on that card.


## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual and unit tests

3. What automated tests I added (or what prevented me from doing so)
Existing and updates tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ravishanker committed:
Update empty state for Most Popular card

ravishanker committed:
Update empty state for latest post summary

ravishanker committed:
Handle Views & Visitors empty state

irfano commented on WordPress/src/main/res/layout/stats_block_text_item.xml:
```diff
@@ -1,19 +1,39 @@
+        app:layout_constraintEnd_toStartOf="@+id/text_icon"
+        tools:text="@string/stats_insights_views_and_visitors_visitors_empty_state" />
+
+    <com.google.android.material.button.MaterialButton
```
We can use `ImageSpan` to add drawable to multiline text in this PR or a new one. It's up to you.

irfano commented on WordPress/src/main/res/layout/stats_block_text_item.xml:
```diff
@@ -1,19 +1,39 @@
         style="@style/StatsBlockText"
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:paddingStart="@dimen/margin_extra_large"
```
No need to add paddingStart for TextView, because the parent `ConstraintLayout` has already padding start `16dp`.

irfano reviewed (changes requested):
We can fix the padding problem, then we're good to merge.

ravishanker committed:
Merge branch 'trunk' into Stats-Revampv2-Empty-states-for-some-cards

ravishanker committed:
Replace icon with ImageSpan

irfano commented on WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/viewholders/TextViewHolder.kt:
```diff
@@ -32,16 +31,30 @@ class TextViewHolder(parent: ViewGroup) : BlockListItemViewHolder(
+                            loadedText.length - 1,
+                            loadedText.length,
+                            Spannable.SPAN_EXCLUSIVE_EXCLUSIVE
+                    )
```
Can we add the drawable instead of replacing `.`? Because this can cause a problem related to the translation process. That happened before.
Internal ref: p1651253093957649/1651250182.578219-slack-C011BKNU1V5

ravishanker committed:
Refactor link handling

Replaced last character with a placeholder to avoid potential issues with translations

irfano reviewed (approved):
LGTM! 👍🏻
